optuna/integration/pytorch_lightning.py

```diff
@@ -69,60 +72,86 @@ def on_fit_start(self, trainer: Trainer, pl_module: "pl.LightningModule") -> Non
                     "optuna.integration.PyTorchLightningPruningCallback"
                     " supports only optuna.storages.RDBStorage in DDP."
                 )
+            # It is necessary to store intermediate values directly in the backend storage because
+            # they are not properly propagated to main process due to cached storage.
+            self._trial.storage.set_trial_system_attr(
+                self._trial._trial_id,
+                _INTERMEDIATE_VALUE,
+                dict(),
+            )
 
     def on_validation_end(self, trainer: Trainer, pl_module: LightningModule) -> None:
-        # When the trainer calls `on_validation_end` for sanity check,
-        # do not call `trial.report` to avoid calling `trial.report` multiple times
-        # at epoch 0. The related page is
+        # Trainer calls `on_validation_end` for sanity check. Therefore, it is necessary to avoid
+        # calling `trial.report` multiple times at epoch 0. For more details, see
         # https://github.com/PyTorchLightning/pytorch-lightning/issues/1391.
         if trainer.sanity_checking:
             return
 
-        epoch = pl_module.current_epoch
-
         current_score = trainer.callback_metrics.get(self.monitor)
         if current_score is None:
             message = (
-                "The metric '{}' is not in the evaluation logs for pruning. "
-                "Please make sure you set the correct metric name.".format(self.monitor)
+                f"The metric '{self.monitor}' is not in the evaluation logs for pruning. "
+                "Please make sure you set the correct metric name."
             )
             warnings.warn(message)
             return
 
+        epoch = pl_module.current_epoch
         should_stop = False
+
+        # Determine if the trial should be terminated in a single process.
+        if not self.is_ddp_backend:
+            self._trial.report(current_score.item(), step=epoch)
+            should_stop = self._trial.should_prune()
+            if not should_stop:
+                return
+            message = f"Trial was pruned at epoch {epoch}."
+            raise optuna.TrialPruned(message)
```

Comment:
Maybe we can simplify the code by removing the local variables.
```suggestion
        if not self.is_ddp_backend:
            self._trial.report(current_score.item(), step=epoch)
            if not self._trial.should_prune():
                return
            raise optuna.TrialPruned(f"Trial was pruned at epoch {epoch}.")
```